fabric-object-builders-v0/src/main/java/net/fabricmc/fabric/api/trading/v1/TradeOfferFactory.java

```diff
@@ -0,0 +1,165 @@
+	}
+
+	/**
+	 * Represents a trade offer where a villager will sell an item or amount of items for some emeralds.
+	 *
+	 * <p>Examples
+	 * <ul>
+	 * <li>3x Emeralds -> 2x White Banners
+	 * <li>1x Emeralds -> 6x Cooked Chicken
+	 * </ul>
+	 *
+	 * @param sellStack The item this villager will sell.
+	 * @param emeraldPrice The amount of emeralds needed to buy this item.
+	 * @param sellCount The amount of the item sold.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @return A new trade offer factory.
+	 */
+	default TradeOffers.Factory createSellItemFactory(ItemStack sellStack, int emeraldPrice, int sellCount, int maxUses, int experience) {
+		return this.createSellItemFactory(sellStack, emeraldPrice, sellCount, maxUses, experience, 0.05F);
+	}
+
+	/**
+	 * Represents a trade offer where a villager will sell an item for amount of items for some emeralds.
+	 *
+	 * <p>Examples
+	 * <ul>
+	 * <li>3x Emeralds -> 2x White Banners
+	 * <li>1x Emeralds -> 6x Cooked Chicken
+	 * </ul>
+	 *
+	 * @param sellStack The item this villager will sell.
+	 * @param emeraldPrice The amount of emeralds needed to buy this item.
+	 * @param sellCount The amount of the item sold.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @param multiplier A price multiplier which represents the rate of price growth when this offer is used.
+	 * @return A new trade offer factory.
+	 */
+	TradeOffers.Factory createSellItemFactory(ItemStack sellStack, int emeraldPrice, int sellCount, int maxUses, int experience, float multiplier);
+
+	/**
+	 * Represents a trade offer where a villager buys an amount of an item for one emerald.
+	 *
+	 * <p>Examples
+	 * <ul>
+	 * <li>1x Oak Boat -> 1x Emerald
+	 * </ul>
+	 *
+	 * @param buyItem The type of item the villager is buying.
+	 * @param buyAmount The amount of items needed to receive one emerald.
+	 * @param maxUses The amount of "uses" a trade can take before a villager needs to restock.
+	 * @param experience The amount of experience the {@link VillagerEntity Villager} will receive.
+	 * @return A new trade offer factory.
+	 */
+	TradeOffers.Factory createBuyItemForOneEmeraldFactory(ItemConvertible buyItem, int buyAmount, int maxUses, int experience);
+
+	/**
+	 * Turns a {@link TradeOffer} into a {@link TradeOffers.Factory}.
+	 *
+	 * <p>Since TradeOffers which currently belong to a villager are mutable, we serialize and then create a new offer to be wrapped.
+	 *
+	 * @param offer The offer to wrap.
+	 * @return A new trade offer factory.
+	 */
+	static TradeOffers.Factory wrap(TradeOffer offer) {
+		return (entity, random) -> new TradeOffer(offer.toTag());
```

Comment:
So do you want to freeze the offer when the factory's create method is called or when this `wrap` method is called? If you freeze on `wrap` method call, the impl should be:
```suggestion
		CompoundTag tag = offer.toTag();
		return (entity, random) -> new TradeOffer(tag);
```
also this method, being static, looks weird out of all other instance methods.

Reply:
I could make a regular wrap method and rename this one to freezeAndWrap

Reply:
`vacuumWrap` and `freezeDry` = `freezeWrap`

Reply:
Decided to take liach's suggestion and freeze before the wrapping process.